src/phosphor-rbmc-data-sync-mgr_main.cpp

```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: Apache-2.0
+
+#include <phosphor-logging/lg2.hpp>
+#include <sdbusplus/async/context.hpp>
+#include <sdbusplus/server/manager.hpp>
+#include <xyz/openbmc_project/DataSync/BMCData/common.hpp>
+
+int main()
+{
+    try
+    {
+        using BMCDataSync =
+            sdbusplus::common::xyz::openbmc_project::data_sync::BMCData;
+
+        sdbusplus::async::context ctx;
+        sdbusplus::server::manager_t objManager{ctx,
+                                                BMCDataSync::namespace_path};
+
+        // clang-tidy currently mangles this into something unreadable
+        // NOLINTNEXTLINE
+        ctx.spawn(
+            [](sdbusplus::async::context& ctx) -> sdbusplus::async::task<> {
+            ctx.request_name(BMCDataSync::interface);
+            co_return;
+        }(ctx));
+
+        ctx.run();
+    }
+    catch (std::exception& e)
+    {
+        lg2::error("Exception: {EXCEPTION}", "EXCEPTION", e);
```

Comment:
this means you'll never get a core dump with a backtrace out of this app.  I think that could be useful though to find the source of the exception?

Reply:
added this commint to PR 1 instead.